cmd/internal/find/primary/meta.go

```diff
@@ -56,4 +59,22 @@ N                   => \d+ (i.e. some number > 0)
 var metaPrimary = Parser.add(&primary{
     tokens: []string{"-meta", "-m"},
     parseFunc: meta.Parse,
+    optionsSetter: func(opts *types.Options) {
+        if !opts.IsSet(types.MaxdepthFlag) {
+            // The `meta` primary's a specialized filter. It should only be used
+            // if a user needs to filter on something that isn't in plugin.EntryAttributes
+            // (e.g. like an EC2 instance tag, a Docker container's image, etc.). Thus, it
+            // wouldn't make sense for `wash find` to recurse when someone's using the `meta`
+            // primary since it is likely that siblings or children will have a different meta
+            // schema. For example, if we're filtering EC2 instances based on a tag, then `wash find`
+            // shouldn't recurse down into the EC2 instance's console output + metadata.json files
+            // because those entries don't have tags and, even if they did, they'd likely be under a
+            // different key (e.g. like "Labels" for Docker containers). Thus to avoid the unnecessary
+            // recursion, we default maxdepth to 1 if the flag was not set by the user. Note that users
+            // who want to recurse down into subdirectories can just specify the maxdepth option, which
+            // is useful when running `wash find` inside a directory whose entries and subdirectory entries
+            // all have the same `meta` schema (e.g. like in an S3 bucket).
+            opts.Maxdepth = 1
```

Comment:
I'm considering whether it's worth allowing something like `wash find -maxdepth` where a no-value `maxdepth` means "recurse." This is useful for people who want to use the `meta` primary and also want `wash find` to recurse all the way down (e.g. it makes the S3 bucket case less annoying). However, implementing this feature would require some extra, not-very-simple code b/c Go's `flag` package requires `int` options to have a value. Thus to avoid the extra code, I wonder if we should add a separate `recurse` option instead and document that the `recurse` option should be used in conjunction with the `meta` primary.

Reply:
Also wondering if I should print something to stderr here so the user knows about this magic.

Reply:
Check for `∞`? I guess the simplest is probably `-maxdepth -1`. Having a separate option feels like it makes it harder to figure out what's happening, but the behavior you're adding does seem useful.

Reply:
I'll go with `-maxdepth -1` -- that makes more sense.